src/Accounts/Authenticators/ClientAssertionAuthenticator.cs

```diff
@@ -45,9 +46,18 @@ public override Task<IAccessToken> Authenticate(AuthenticationParameters paramet
                 AuthorityHost = new Uri(authority),
                 TokenCachePersistenceOptions = spParameters.TokenCacheProvider.GetTokenCachePersistenceOptions()
             };
+
             options.DisableInstanceDiscovery = spParameters.DisableInstanceDiscovery ?? options.DisableInstanceDiscovery;
             options.Diagnostics.IsTelemetryEnabled = false; // disable telemetry to avoid error thrown from Azure.Core that AssemblyInformationalVersion is null
             TokenCredential tokenCredential = new ClientAssertionCredential(tenantId, spParameters.ClientId, () => GetClientAssertion(spParameters), options);
+
+            base.CollectTelemetry(tokenCredential);
+            CheckTokenCachePersistanceEnabled = () =>
```

Comment:
Just questioning - If `CheckTokenCachePersistanceEnabled` is implemented in most authenticators, can we define this function in a more common place?

Reply:
CheckTokenCachePersistanceEnabled is a function variable, but its definition varies in different calling places.